docs/source/user_guide/pipeline-components.md

```diff
@@ -50,18 +50,186 @@ Component details and demo pipelines can be found in the `https://github.com/ely
+```
+
+#### Adding components to the registry
+
+To add a component registry entry run `elyra-metadata install component-registries`.
+
+```bash
+$ elyra-metadata install component-registries \
+       --display_name="filter components" \
+       --description="filter text in files" \
+       --runtime=kfp \
+       --location_type=URL \
+       --paths="['https://raw.githubusercontent.com/elyra-ai/elyra/master/etc/config/components/kfp/filter_text_using_shell_and_grep.yaml']" \
+       --categories='["filter file content"]'
+```
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+#### Modifying a component registry entry
+
+To replace a component registry entry run `elyra-metadata install component-registries` and specify the `--replace` option:
+
+```bash
+$ elyra-metadata install component-registries \
+       --name="filter_components" \
+       --display_name="filter components" \
+       --description="filter text in files" \
+       --runtime=kfp \
+       --location_type=URL \
+       --paths="['https://raw.githubusercontent.com/elyra-ai/elyra/master/etc/config/components/kfp/filter_text_using_shell_and_grep.yaml']" \
+       --categories='["file operations"]' \
+       --replace
+```
+
+Note: You must specify all property values, not only the ones that you want to modify.
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+#### Deleting components from the registry
+
+To delete a component registry entry and its component definitions:
+
+```bash
+$ elyra-metadata remove component-registries \
+       --name="filter_components"
+```
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+### Configuration properties
+
+The component registry entry properties are defined as follows. The string in the headings below, which is enclosed in parentheses, denotes the CLI option name.
+
+##### Name (display_name)
+
+A user-friendly name for the registry entry. Note that the registry entry name is not displayed in the palette. This property is required.
+
+Example: `data load components`
+
+##### N/A (name)
+
+A unique identifier for this registry entry. A value is generated from `Name` if no value is provided.
+
+Example: `data_load_components`
+
+##### Description (description)
+
+A description for the registry entry.
+
+Example: `Load data from external data sources`
+
+##### Category (categories)
+
+In the pipeline editor palette components are grouped into categories to make them more easily accessible. If no category is provided, the components defined by this registry entry are added to the palette under `no category`.  
+
+Examples (CLI):
+
+- `['load data from database']`
+- `['train model','pytorch']`
+
+
+##### Runtime (runtime)
+
+The runtime environment that supports the component(s). Valid values are `kfp` and `airflow`. This property is required.
```

Comment:
This doesn't account for other runtimes or uninstalled runtimes.  I think we should state something like: _Valid values are the set of currently configured runtimes that appear in the dropdown (UI) or help-text (CLI)._

Reply:
Changed! That said (off-topic), I do believe the CLI should also provide a command that lists supported runtimes (user friendly name and  value to be specified). (Help is still hard to use)

Example (ignoring for now semantic ambiguity because elyra-metadata uses `runtimes` to refer to runtime configurations:

```
$ elyra-pipeline list runtimes
   runtime id       runtime name
   ---------------------------------
   kfp                   Kubeflow Pipelines
   airflow             Apache Airflow 1.x
```